### ISSUE TITLE
Accessing read-only paths

### ISSUE THREAD

**Issue by PassiveLemon** [AdGoBye/AdGoBye#104]
I am currently attempting to package this in Nix but upon running the program, it attempts to write to the read-only nix store `/nix/store/258d5k9hcdqp79m6izfzh6936lmf3j1l-adgobye-4.0.2+v4IsBrokenPullMe/lib/adgobye/Plugins`

Shouldn't plugin data be managed in the users .config or .local directory?

**Comment by regalialong** [comment]
Err, yes, AdGoBye expects to have all its files in the same folder as the binary. If I made it a first class Linux app, that probably should go into .local and config instead.

I'm not sure about the filesystem restrictions of NixOS, else I'd be fine with doing a patch to allow picking the working directory for the app.

**Comment by PassiveLemon** [comment]
It's pretty much packages are built and put into the nix store (/nix/store) which is read-only so any binary that does things based on the location of the binary will probably have problems. There are probably ways to fix this on the Nix side but I think this is an appropriate upstream issue anyways

**Comment by regalialong** [comment]
Yeah no, completely valid, I'll take a look at it when I find the time.

**Comment by regalialong** [comment]
Do you mind testing the artifact for https://github.com/AdGoBye/AdGoBye/actions/runs/11190863213? I did this fairly lazily (I'm tired and not really into it today) but it should be good enough if I understand correctly.

Make sure there's no `appsettings.json` next to the binary, and either start it once and let it create `$XDG_DATA_HOME/AdGoBye` for you or create it yourself and move `appsettings.json` into the folder.

**Comment by PassiveLemon** [comment]
I'll give it a test tomorrow

**Comment by PassiveLemon** [comment]
I'm actually building it from source because it's a little easier than fetching a binary.
When building, it fails with these errors:
```
/build/adgobye/AdGoBye/Program.cs(32,32): error CS0117: 'Log' does not contain a definition for 'Log' [/build/adgobye/AdGoBye/AdGoBye.csproj]
/build/adgobye/AdGoBye/PluginLoader.cs(105,23): error CS0117: 'Settings' does not contain a definition for 'Options' [/build/adgobye/AdGoBye/AdGoBye.csproj]
/build/adgobye/AdGoBye/PluginLoader.cs(110,17): error CS0122: 'Logger' is inaccessible due to its protection level [/build/adgobye/AdGoBye/AdGoBye.csproj]
```
v4.0.2 builds fine

**Comment by regalialong** [comment]
My bad, I think these were usually obscured by the debug macros, I'm honestly not sure how I didn't catch that sooner.
9ef438f4bffdc032ea7dc7d1fa7c01f5527ce21b should fix that

**Comment by PassiveLemon** [comment]
It appears to run now! I haven't tested but it mentions processing worlds at my install dir for VRChat so that seems like a good sign. Could the appsettings.json be created if it doesn't already exist? That's all I can think of

**Comment by regalialong** [comment]
Yeah, I think we can embed appsettings.json as a embedded Resource and use that for bootstrapping, we get access to resources per `Assembly.GetManifestResourceStream`.
I'll look into it when I feel a bit more like touching AGB again, this was definitely just the lazy unbreak-it-now implementation for now.
